R/unnest-longer.R

```diff
@@ -194,21 +205,23 @@ col_to_long <- function(col,
 # Convert a list element to a long tibble with:
 # - 1 col (2 if `indices_include` is `TRUE`)
 # - N rows, where `N = vec_size(x)`
+#   - If `keep_empty == TRUE` and `vec_size(x) == 0`, then `N = 1` instead
 elt_to_long <- function(x,
                         index,
                         name,
                         values_to,
                         indices_to,
                         indices_include,
+                        keep_empty,
                         error_call = caller_env()) {
   if (is.null(x)) {
-    x <- unspecified(1L)
+    x <- unspecified()
 
     if (indices_include) {
       if (is.integer(index)) {
-        index <- 1L
+        index <- integer()
       } else {
-        index <- NA_character_
+        index <- character()
```

Comment:
General idea is to have 1 standard path we always apply to `NULL`, and then if `keep_empty = TRUE` then the other path below handles both the updated `NULL` result we created here and the empty vector case